src/lib/tvgPictureImpl.h

```diff
@@ -34,43 +34,72 @@ struct Picture::Impl
 {
     unique_ptr<Loader> loader = nullptr;
     Paint* paint = nullptr;
+    uint32_t *pixels = nullptr;
+    Picture *picture = nullptr;
+    void *edata = nullptr;              //engine data
+    uint32_t flag = RenderUpdateFlag::None;
 
-    bool dispose(RenderMethod& renderer)
+    Impl(Picture* p) : picture(p)
     {
-        if (!paint) return false;
+    }
 
-        paint->pImpl->dispose(renderer);
-        delete(paint);
+    bool dispose(RenderMethod& renderer)
+    {
+        if (paint) {
+            paint->pImpl->dispose(renderer);
+            delete(paint);
 
-        return true;
+            return true;
+        }
+        else if (pixels) {
+            return renderer.dispose(edata);
+        }
+        return false;
     }
 
     void reload()
     {
-        if (loader && !paint) {
-            auto scene = loader->data();
-            if (scene) {
-                paint = scene.release();
-                if (!paint) return;
-                loader->close();
+        if (loader) {
+            if (!paint) {
+                auto scene = loader->scene();
+                if (scene) {
+                    paint = scene.release();
+                    if (!paint) return;
+                    loader->close();
+                }
+            }
+            if (!pixels) {
+                pixels = (uint32_t*)loader->pixels();
             }
         }
     }
 
-
-    void* update(RenderMethod &renderer, const RenderTransform* transform, uint32_t opacity, vector<Composite>& compList, RenderUpdateFlag flag)
+    void* update(RenderMethod &renderer, const RenderTransform* transform, uint32_t opacity, vector<Composite>& compList, RenderUpdateFlag pFlag)
     {
         reload();
 
-        if (!paint) return nullptr;
-
-        return paint->pImpl->update(renderer, transform, opacity, compList, flag);
+        if (pixels) {
```

Comment:
1. better idea:
bool flag = reload();

less LOC

if (pixels) edata = ;
else  edata = ;

return edata;

Reply:
reload() called every frame. 
We want to pass RenderUpdateFlag::Image or All to prepare only once after load() is called.
So, in my opinion, we need the flag member variable.
Also, reload() is being called on duplicate as well, but flag information is not required.

I don't think this change is absolutely necessary.

Reply:
please double-check it.

Reply:
i have changed code